Sources/MapboxMaps/Style/FormattedElement.swift

```diff
@@ -0,0 +1,43 @@
+public typealias Formatted = [FormattedElement]
+
+public enum FormattedElement: Codable {
+    case format
+    case substring(String)
+    case formatOptions(FormatOptions)
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+
+        if let validString = try? container.decode(String.self), validString == Expression.Operator.format.rawValue {
+            self = .format
+            return
+        }
+
+        if let validString = try? container.decode(String.self) {
+            self = .substring(validString)
+            return
+        }
+
+        if let validOptions = try? container.decode(FormatOptions.self) {
+            self = .formatOptions(validOptions)
+            return
+        }
+
+        let context = DecodingError.Context(codingPath: decoder.codingPath,
+                                            debugDescription: "Failed to decode Formatted")
+        throw DecodingError.dataCorrupted(context)
```

Comment:
consider using `if`/`else if`/`else` here to reduce the likelihood that we refactor this in the future and forget to `return` from one of these if blocks.